### PR TITLE
Reason description for ignored tests

### PR DESCRIPTION
**Pull request by peter-gerhard**
@agourlay pls review
I went with `_ignoredBecause` field and `ignoredBecause` method for the builder. Let me know if you prefer a different style :)

### CODE REVIEW

File: cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/Dsl.scala [hunk 1]
```diff
@@ -23,11 +23,23 @@ import scala.concurrent.duration.FiniteDuration
...
-    BodyElementCollector[ScenarioDef, FeatureDef](scenarios ⇒ FeatureDef(name, scenarios, ignored))
+  def Feature(name: String) = FeatureBuilder(name)
 
-  def Scenario(name: String, ignored: Boolean = false) =
-    BodyElementCollector[Step, ScenarioDef](steps ⇒ ScenarioDef(name, steps, ignored))
+  case class FeatureBuilder(name: String, _ignoredBecause: Option[String] = None) {
+    def ignoredBecause(reason: String) = copy(_ignoredBecause = Some(reason))
+  }
+
+  implicit def featureBuilder(f: FeatureBuilder): BodyElementCollector[ScenarioDef, FeatureDef] =
```
Comment (agourlay) [inline cornichon-core/src/main/scala/com/github/agourlay/cornichon/dsl/Dsl.scala]: I have not thought about the integration with the macro, nicely done 👍 

I guess those dedicated builders might be useful in the future to add more knobs on the ```Scenario``` and ```Feature```.

**Comment by agourlay** [comment]
Looks great! 🎉 

Would be awesome if you could update the docs as well 
https://github.com/agourlay/cornichon/blob/master/cornichon-docs/src/main/tut/feature-options.md#ignoring-features-or-scenarios

**Comment by peter-gerhard** [comment]
🙃 forgot about that f446b94

**Comment by agourlay** [comment]
The test are failing because ```ignored``` is used there and there is a reason in the comments :)

https://travis-ci.org/agourlay/cornichon/jobs/243199447#L2775

**Comment by peter-gerhard** [comment]
ahh, the ol `test:compile` vs `compile` ...
Now it should pass.

I made a few changes again. 
1) i was unhappy that the IDE would show the `_,ignoredBecause` parameter (even before the method)
2) users shouldn't be able to instantiate the builders directly
3) since the reason should not show up anywhere else then in the feature file yet I removed it from the model again so that it is only part of the dsl

pls have another look

**Comment by peter-gerhard** [comment]
sry i cant make out out why the build is failing, locally all tests pass :(

**Comment by agourlay** [comment]
Those errors are not linked to your PR. 

The GraphQL testing is super flaky on Travis (I get it locally as well when my computer is already under heavy load)

```
[info] - should demonstrate GraphQL support *** FAILED ***

[info]   Scenario 'demonstrate GraphQL support' failed:

[info]   

[info]   at step:

[info]   When I query GraphQL endpoint /graphql with query 

[info]               query($sessionId: String!) {

[info]                superheroByName(sessionId: $sessionId, name: "Batman", protectIdentity: false) {

[info]                  name

[info]                  city

[info]                }

[info]               }

[info]              and with variables sessionId -> "uwIgBmCn"

[info]   

[info]   with error(s):

[info]   HTTP POST request to http://localhost:8080/graphql

[info]   without parameters

[info]   without headers

[info]   with body

[info]   {

[info]     "query" : "\n            query($sessionId: String!) {\n             superheroByName(sessionId: $sessionId, name: \"Batman\", protectIdentity: false) {\n               name\n               city\n             }\n            }\n          ",

[info]     "operationName" : null,

[info]     "variables" : {

[info]       "sessionId" : "uwIgBmCn"

[info]     }

[info]   }

[info]   connection timed out error after 2000 ms

[info]   

[info]   replay only this scenario with the command:

[info]   testOnly *SuperHeroesScenario -- -t "Cornichon feature example should demonstrate GraphQL support" (ScalatestFeature.scala:57)
```

I will have to look at it eventually!

The PR LGTM 👍

**Pull request merged**